pkg/webui/console/views/device-add/wizard/network-settings-form/validation-schema.js

```diff
@@ -21,7 +21,13 @@ const validationSchema = Yup.object({
   frequency_plan_id: Yup.string().required(sharedMessages.validateRequired),
   lorawan_version: Yup.string().required(sharedMessages.validateRequired),
   lorawan_phy_version: Yup.string().required(sharedMessages.validateRequired),
-  supports_class_c: Yup.boolean().default(false),
+  supports_class_c: Yup.boolean().when(['$activationMode'], (mode, schema) => {
+    if (mode === ACTIVATION_MODES.MULTICAST) {
+      return schema.oneOf([true], sharedMessages.validateRequired)
+    }
+
+    return schema.default(false)
+  }),
```

Comment:
This needs to work in a way that the `Supports Class C` checkbox will be checked by default while the checkbox will be disabled. Please also add a field description in these cases that the device has to support class c when using multicast.

Reply:
> Please also add a field description in these cases that the device has to support class c when using multicast.

This is technically incorrect. Multicast devices must support either class b or c, we currently do not have functionality to create class b compatible devices via the console, but it is possible via the CLI. Support for class B will be added in https://github.com/TheThingsNetwork/lorawan-stack/pull/3065 as it requires fields from `mac_settings`.
Adding description that a multicast device _has to support class c_ raises more confusion for both experienced and novice users than adds any value.

Currently, the problem is that the end device wizard has option to register multicast devices but it is not working. With this PR merged we at least can allow creating multicast devices via the console by allowing only class c compatible devices regardless of https://github.com/TheThingsNetwork/lorawan-stack/pull/3065 being merged within the same release as this fix or in the next release.

Reply:
In any case, I'd prefer a solution that does not force the user to click a checkbox to proceed. If we know this option has to be checked then we should preselect that value and prevent changes to the field by disabling it. The confusion you mention will occur just as much when the user is faced with a generic validation message that the field is required.

Anyway, since the logic will be changed soon anyway via #3065, it's not worth spending more time on perfecting this now.